### PR TITLE
Start adding modern chialisp

### PR DESCRIPTION
**Pull request by prozacchiwawa**
This adds a page detailing modern chialisp with several examples.

### CODE REVIEW

File: docs/modern-chialisp.md [hunk 1]
```diff
@@ -0,0 +1,1438 @@
...
+programs and embedding their compiled representation.  Since chialisp programs
+contain identification about how they should be compiled, it's possible to
+include programs from different versions of the language accurately.
+
+```chialisp
+(mod (Z)
+    
+  (include *standard-cl-23*)
+    
+  (embed-file hello-data bin "hello.txt")
```
Comment (trepca) [inline docs/modern-chialisp.md:319]: how about compile-include forms? is there a different syntax for that?

Comment (prozacchiwawa) [reply docs/modern-chialisp.md:319]: Hasn't been merged yet, getting there.

Comment (prozacchiwawa) [reply docs/modern-chialisp.md:319]: Thanks, will add.

**Review by BrandtH22 — APPROVED** [review]
lgtm

**Pull request merged**